src/messages/mod.rs

```diff
@@ -78,19 +78,50 @@ impl HopMessage {
     }
 }
 
+/// Metadata needed for verification of the sender
+#[derive(Ord, PartialOrd, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
+pub struct FullSecurityMetadata {
+    proof: SectionProofChain,
+    sender_prefix: Prefix<XorName>,
+    signature: BlsSignature,
+}
+
+impl FullSecurityMetadata {
+    pub fn verify_sig(&self, bytes: Vec<u8>) -> bool {
```

Comment:
Why do we need to move bytes into this function?

Reply:
A good question - it's quite possible that we don't :+1: